modules/auto-update/implementation/ot-auto-updater.js

```diff
@@ -0,0 +1,240 @@
+     * @param {String} config.tempLocation - The local dir to save temporary information for Auto Git Update.
+     * @param {String} config.executeOnComplete - A command to execute after an update completes. Good for restarting the app.
+     */
+    constructor(config) {
+        this.config = config;
+    }
+
+    initialize(logger) {
+        if (!this.config) throw new Error('You must pass a config object to AutoUpdater.');
+        if (!this.config.branch) this.config.branch = 'master';
+        if (!this.config.tempLocation) throw new Error('You must define a temp location for cloning the repository');
+        this.logger = logger;
+    }
+
+    /**
+     * Checks local version against the remote version and updates if different. 
+     */
+    async autoUpdate() {
+        const versionCheck = await this.compareVersions();
+        if (versionCheck.upToDate) {
+            return true
+        };
+        const update = await this.forceUpdate();
+        return update;
+    }
+
+    /**
+     * @typedef VersionResults
+     * @param {Boolean} UpToDate - If the local version is the same as the remote version.
+     * @param {String} currentVersion - The version of the local application.
+     * @param {String} remoteVersion - The version of the application in the git repository. 
+     * 
+     * Checks the local version of the application against the remote repository.
+     * @returns {VersionResults} - An object with the results of the version comparison.
+     */
+    async compareVersions() {
+        try {
+            this.logger.info('AutoUpdater - Comparing versions...');
+            const currentVersion = this.readAppVersion();
+            const remoteVersion = await this.readRemoteVersion();
+            this.logger.info(`AutoUpdater - Current version: ${currentVersion}`);
+            this.logger.info(`AutoUpdater - Remote Version: ${remoteVersion}`);
+            if (currentVersion === remoteVersion) {
+                return {
+                    upToDate: true, 
+                    currentVersion
+                };
+            };
+            return {
+                upToDate: false, 
+                currentVersion, 
+                remoteVersion
+            };
+        }catch(e) {
+            this.logger.error(`AutoUpdater - Error comparing local and remote versions. Error message: ${e.message}`);
+            return {
+                upToDate: false, 
+                currentVersion: 'Error', 
+                remoteVersion: 'Error'
+            }
+        }
+    }
+
+    /**
+     * Clones the git repository and installs the update over the local application.
+     * A backup of the application is created before the update is installed.
+     * If configured, a completion command will be executed and the process for the app will be stopped. 
+     * @returns {Boolean} The result of the update.
+     */
+    async forceUpdate() {
```

Comment:
Why is this forced update? I guess that we can rename this method to be just: update or runUpdate